sopel/builtins/xkcd.py

```diff
@@ -26,19 +26,21 @@
 # https://twitter.com/Dmdboi/status/1589202274999767041
 SEARCHXKCD_API = 'https://gq67pznq1k.execute-api.eu-west-1.amazonaws.com/search'
 
-ignored_sites = [
-    # For searching the web
-    'almamater.xkcd.com',
-    'blog.xkcd.com',
-    'blag.xkcd.com',
-    'forums.xkcd.com',
-    'fora.xkcd.com',
-    'forums3.xkcd.com',
-    'store.xkcd.com',
-    'wiki.xkcd.com',
-    'what-if.xkcd.com',
-]
-sites_query = ' site:xkcd.com -site:' + ' -site:'.join(ignored_sites)
+
+class SearchXkcdError(Exception):
+    """Generic exception to raise if there was a problem contacting the API."""
+
+
+class SearchFailedError(SearchXkcdError):
```

Comment:
Nitpick: That seems a little specialized to me, I would probably have used the base exception to cover this.

Reply:
Maybe it needs a better name? The idea was to shadow `requests.exceptions.ConnectionError` with a plugin-specific error type—note where `SearchFailedError` is used below. It's an attempt to differentiate between errors of the "I talked to the API but couldn't understand what it said back" kind and more serious "the API might be gone???" issues.

I might look at this again when I go through to fix up things like your logging comment and reevaluate the exception hierarchy, even though you did say "LGTM".